agent/engine/task_manager.go

```diff
@@ -524,23 +524,29 @@ func (mtask *managedTask) emitTaskEvent(task *apitask.Task, reason string) {
 	seelog.Infof("Managed task [%s]: sent task change event [%s]", mtask.Arn, event.String())
 }
 
-// emitContainerEvent passes a given event up through the containerEvents channel if necessary.
-// It will omit events the backend would not process and will perform best-effort deduplication of events.
-func (mtask *managedTask) emitContainerEvent(task *apitask.Task, cont *apicontainer.Container, reason string) {
-	event, err := api.NewContainerStateChangeEvent(task, cont, reason)
+// emitManagedAgentEvent passes a special task event up through the taskEvents channel if there are managed
+// agent changes in the container passed as parameter.
+// It will omit events the backend would not process
+func (mtask *managedTask) emitManagedAgentEvent(task *apitask.Task, cont *apicontainer.Container, reason string) {
```

Comment:
I don't see the changes for emitting a stopped status event for managed agent? is that out of scope of this PR?

Reply:
It is out of scope of the current PR.